service/storage_service.cc

```diff
@@ -4011,5 +4011,14 @@ future<> storage_service::send_node_ops_cmd_to_node(const node_ops_cmd_response_
     });
 }
 
+void storage_service::register_node_operation_observer(node_operations_observer observer) {
+        _node_operations_observers.emplace_back(std::move(observer));
+}
+
+future<> storage_service::notify_observers(const node_ops_cmd_request& req) {
+    return parallel_for_each(_node_operations_observers, [&req, this] (node_operations_observer& observer) {
+        return observer(req, *this);
+    });
```

Comment:
I don't know if the point where we send/receive the command is the right one. We already have node status (normal/joining/leaving), perhaps we should observe where the state is maintained, not changed. This is more resilient against failures, and against new APIs.

Reply:
/cc @asias

Reply:
I think it is a bad idea to depend on the new node ops cmd infrastructure to do something when a node is decommissioned. 

We already have ways to listen for a node is decommissioned. We can use endpoint_lifecycle_subscriber::on_leave_cluster.

If you do not like the notifier, you can even check the gossip status of the node directly in your loop. For instance take a look: storage_service::restore_replica_count.

Also, should we take care of node that is removed by  `nodetool removenode`. With on_leave_cluster, both cases are handled. Maybe you care less about removenode because the node is DOWN anyway.

Reply:
OK sure.
I guess that the subscriber is what I was looking for. I just missed it when I searched for a way to get notified.